symphony-bdk-core/src/main/java/com/symphony/bdk/core/auth/OAuthSession.java

```diff
@@ -0,0 +1,47 @@
+package com.symphony.bdk.core.auth;
+
+import com.symphony.bdk.core.auth.exception.AuthUnauthorizedException;
+import com.symphony.bdk.http.api.ApiException;
+
+import lombok.RequiredArgsConstructor;
+import lombok.extern.slf4j.Slf4j;
+import org.apiguardian.api.API;
+
+import java.time.Instant;
+
+@Slf4j
+@RequiredArgsConstructor
+@API(status = API.Status.INTERNAL)
+public
+class OAuthSession {
+
+  private final AuthSession authSession;
+
+  public String getBearerToken() throws ApiException {
+    try {
+      refreshIfNeeded();
+    } catch (AuthUnauthorizedException e) {
+      throw new ApiException(401, e.getMessage());
+    }
+    return authSession.getAuthorizationToken();
+  }
+
+  /**
+   * If the bearer token is expired this function is going to call the login endpoint and
+   * update the auth session with new tokens.
+   *
+   * @throws AuthUnauthorizedException if the authentication fails
+   */
+  private void refreshIfNeeded() throws AuthUnauthorizedException {
+    int interval = 5;
```

Comment:
`leeway` could be a more meaningful variable name. And it could be put as a static final field.

Reply:
https://github.com/finos/symphony-bdk-java/pull/618/commits/c7cd1cfe370a57d178dab78a22de227d5e199292